src/devices/Charlieplex/CharlieplexSegment.cs

```diff
@@ -210,5 +213,88 @@ public void Dispose()
                 _gpioController = null!;
             }
         }
+
+        // IOutputSegment Implementation
+        // Only supported when shift register is connected with GPIO
+
+        /// <summary>
+        /// The length of the segment; the number of GPIO pins it exposes.
+        /// </summary>
+        int IOutputSegment.Length => _segment.Length;
```

Comment:
Do you mean they are `Interface.` or why I'm relying on `VirtualOutputSegment` for the implementation?

Reply:
Yeah, the way they are right now you need to cast the class to the interface in order to call the methods which is less than ideal. Meaning:

```c#
CharlieplexSegment myCPS = new CharlieplexSegment();
var theLength = myCPS.Length; // <-- This line won't compile
var theLength = ((IOutputSegment)myCPS).Length; // <-- This line does compile
```

Reply:
To fix, simply change these to look like:
```suggestion
        public int Length => _segment.Length;
```

Reply:
That's what I thought you meant. I did it like that because each of the APIs have their own concepts and I didn't want this set of APIs to get confused with those. For this one (`Length`) however, it might make sense to make it generic. For `ShiftRegister`, I went with `BitLength` originally. I don't recall why. I'll make this change to `Length` on each of the implementations.